### PR TITLE
Fixing ergonomics of TextureData and TextureMetadata

### PR DESCRIPTION
**Pull request by Rhuagh**

### CODE REVIEW

File: amethyst_renderer/src/formats/texture.rs [hunk 1]
```diff
@@ -57,6 +108,12 @@ pub enum TextureData {
     U64(Vec<u64>, Option<TextureMetadata>),
 }
 
+impl From<[f32; 4]> for TextureData {
+    fn from(color: [f32; 4]) -> Self {
+        TextureData::Rgba(color, None)
```
Comment (torkleyy) [inline amethyst_renderer/src/formats/texture.rs]: Maybe we could not wrap metadata in `Option` anymore?

**Review by torkleyy — COMMENTED** [review]
Looks great, thank you!

File: amethyst_renderer/src/formats/texture.rs (hunk 1 above)
Comment (Rhuagh) [reply amethyst_renderer/src/formats/texture.rs]: Possibly, I thought about it. I'll remove the Option, users can just use Default instead

**Comment by Rhuagh** [comment]
Updated.

**Review by torkleyy — APPROVED** [review]
Great, thank you!

**Pull request merged**